proxy/logging/LogFilter.cc

```diff
@@ -465,112 +465,60 @@ LogFilterString::display(FILE *fd)
   LogFilterInt::LogFilterInt
   -------------------------------------------------------------------------*/
 
-void
-LogFilterInt::_setValues(size_t n, int64_t *value)
+LogFilterInt::LogFilterInt(LogFilterInt const &that) : LogFilter(that.m_name, that.m_field, that.m_action, that.m_operator)
 {
   m_type       = INT_FILTER;
-  m_num_values = n;
-  if (n) {
-    m_value = new int64_t[n];
-    memcpy(m_value, value, n * sizeof(int64_t));
-  }
+  m_value      = that.m_value;
+  m_num_values = m_value.size();
 }
 
-// TODO: ival should be int64_t
 int
-LogFilterInt::_convertStringToInt(char *value, int64_t *ival, LogFieldAliasMap *map)
+LogFilterInt::_convertStringToInt(ts::TextView val, int64_t *ival, LogFieldAliasMap *map)
```

Comment:
Wow, these are some confusing APIs, "convert to int" returns an int, which is a flag (so why not bool?), and it stores the "int" value in an int64_t pointer. What were we smoking, and can I have some?

Out of scope here, but I kinda wish we'd standardize on specific int sizes, like int64_t, int32_t, intu64_t etc.

Reply:
I can clean that up a bit, too. It does need to return two values, a flag to indicate if it was a well formed integer, and the actual integer value. I can change that to return a 2-tuple, which is the C++17 way of handling it. That will make it different from the other filter logic.

Reply:
How's that?